tests/sys/zuse/dawn.hoon

```diff
@@ -0,0 +1,265 @@
+/+  *test
+|%
+:: example hull for ~zod
+::
+++  hul
+  ^-  hull:constitution:ethe
+  :+  [0x0 0x0 0x0 0x0]
+    :*  ~
+        life=1
+         pass=2.448.360.348.730.164.860.814.441.775.703.143.856.915.192.920.
+        639.124.529.297.987.279.849.833.790.775.864.413.949.853.880.667.744.
+        188.597.545.066.664.466.963.044.328.182.155.965.137.512.758.548.384.
+        637.214.562
+        continuity-number=0
+        sponsor=[~ u=~zod]
+        escape=~
+    ==
+  [~ u=[spawn-proxy=0x0 spawn-count=2 spawned=~]]
+:: secret key for ~zod
+++  sec
+  ^-  ring
+  0w8.Ugyke.eUOf2.NcHRo.tZA7r.P8vP6.DGKp4.yn-BI.etdzb.ucv75.WgRob.H1-7n.
+  4qCje.gc7z7.1i-3T.9~8UR.IGkGH.6NWR2
+:: this produces a 1000+ line payload, so we just check that it doesn't crash
+::
+++  test-give-czar
+  =/  zar  czar:give:dawn
+  ~!  zar
+  %+  expect-eq
+    !>  &
+    !>  ?=(^ zar)
+::
+++  test-give-hull
+  =/  oct
+    %-  as-octs:mimes:html
+    %+  rap  3
+    :~  '{"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0xb322069400'
+        '00000000000000000000000000000000000000000000000000000000000000","to"'
+        ':"0x84b37fbc6188da8ae8661eae322af4d92db45ecc"},"latest"]}'
+    ==
+  %+  expect-eq
+    !>  oct
+    !>  (hull:give:dawn ~zod)
+::
+++  test-give-turf
+  =/  oct
+    %-  as-octs:mimes:html
+    %+  rap  3
+    :~  '[{"jsonrpc":"2.0","id":"0","method":"eth_call","params":[{"data":"0x'
+        'eccc8ff1000000000000000000000000000000000000000000000000000000000000'
+        '0000","to":"0x84b37fbc6188da8ae8661eae322af4d92db45ecc"},"latest"]},'
+        '{"jsonrpc":"2.0","id":"1","method":"eth_call","params":[{"data":"0xe'
+        'ccc8ff10000000000000000000000000000000000000000000000000000000000000'
+        '001","to":"0x84b37fbc6188da8ae8661eae322af4d92db45ecc"},"latest"]},{'
+        '"jsonrpc":"2.0","id":"2","method":"eth_call","params":[{"data":"0xec'
+        'cc8ff100000000000000000000000000000000000000000000000000000000000000'
+        '02","to":"0x84b37fbc6188da8ae8661eae322af4d92db45ecc"},"latest"]}]'
+    ==
+  %+  expect-eq
+    !>  oct
+    !>  turf:give:dawn
+::
+++  test-take-czar
+  =/  oct
+    %-  as-octs:mimes:html
+    %+  rap  3
+    :~  '[{"id":"0","jsonrpc":"2.0","result":"0xb69b6818b17b7cc22f8e0a2291f58'
```

Comment:
For these big result blobs, maybe consider splitting them as 32 bytes per line, for readability of the test data.

Reply:
You wouldn't prefer to maximize the available 80 chars? I'm not sure anything will look good here ...

Worth noting, several those blobs contains the Ships contract address, so they'll need to change at least once more.

Reply:
32 bytes as in 64 characters, and we already use 8 for left-hand padding. I think being able to comfortably read the data is more valuable than 8 "wasted" characters.  
To clarify, I mean to say that doing 32 bytes per line allows you to separate the arguments to be actually readable. Everything but variable-length data is just a 32 byte line.

Ships contract address should maybe be interpolated in from zuse then?

Reply:
I mean if the goal is to read the data, you probably want to construct a hoon json literal and compare against _that_.

Reply:
Reformatted.

@ohAitch, this goal here is to completely encapsulate en/de-serialization, so vere only needs to only make very straightforward function calls. I'll tag you in when I open that PR.